### PR TITLE
Remove abbreviated displayname 

### PR DESCRIPTION
**Pull request by basine**
The UI won't display the abbreviated displayName anymore. I took this chance to rewrite the displayNameGenerator and related classes in Swift. This however leads to a performance drop due to Swift - ObjectiveC bridging of String / NSString :-/

### CODE REVIEW

File: Source/Model/User/DisplayNameGenerator.swift [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+public class DisplayNameGenerator : NSObject {
+    public var idToPersonNameMap : [NSManagedObjectID: PersonName] = [:]
```
Comment (daehn) [inline Source/Model/User/DisplayNameGenerator.swift]: It looks like this property could be declared as `private`.

File: Source/Model/User/DisplayNameGenerator.swift [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        mapUsersToNames(returnChanges: false)
+    }
+    
+}
+
+let DisplayNameGeneratorKey = "DisplayNameGenerator"
+
+extension NSManagedObjectContext {
+    
+    var nameGenerator : DisplayNameGenerator? {
```
Comment (daehn) [inline Source/Model/User/DisplayNameGenerator.swift]: Does this object exist in both contexts? If not an assert could be added to ensure it is accessed from the correct thread.

File: Tests/Source/Model/User/ZMPersonNameTests.m [hunk 3]
```diff
@@ -18,15 +18,15 @@
 
 
 #import "ZMBaseManagedObjectTest.h"
-#import "ZMPersonName.h"
-#import "NSString+ZMPersonName.h"
+//#import "PersonName.h"
```
Comment (daehn) [inline Tests/Source/Model/User/ZMPersonNameTests.m]: Can be removed?

File: Source/Model/User/DisplayNameGenerator.swift (hunk 2 above)
Comment (basine) [inline Source/Model/User/DisplayNameGenerator.swift]: The displayNameGenerator is set when setting up the uiMOC. Adding an assert makes a lot of tests crash that use `performPretendingUiMocIsSyncMoc` :-/

**Comment by daehn** [comment]
💣 Conflicting

**Pull request merged**